src/gwt/src/org/rstudio/core/client/command/AppCommand.java

```diff
@@ -100,19 +114,79 @@ public void onVisibleChanged(AppCommand command)
          parentToolbar_.invalidateSeparators();
       }
 
-      private final AppCommand command_;
+      protected final AppCommand command_;
       private boolean synced_ = true;
-      private HandlerRegistration handlerReg_;
-      private HandlerRegistration handlerReg2_;
-      private Toolbar parentToolbar_;
+      protected HandlerRegistration handlerReg_;
+      protected HandlerRegistration handlerReg2_;
+      protected Toolbar parentToolbar_;
+   }
+
+   private class CommandSourceColumnToolbarButton extends CommandToolbarButton
```

Comment:
It feels a little weird that the `AppCommand` knows so much about source columns (these concepts should generally not be coupled except where necessary). What do you think about moving the source column bits into a separate package? You can could something like a `SourceColumnCommand` which wraps an AppCommand and adds the ability to generate the more contextual enable/disable states.